src/pages/groupComparison/GroupComparisonStore.ts

```diff
@@ -276,15 +278,111 @@ export default class GroupComparisonStore extends ComparisonStore {
         },
     });
 
+    readonly allStudies = remoteData(
+        {
+            invoke: async () =>
+                await client.getAllStudiesUsingGET({
+                    projection: 'SUMMARY',
+                }),
+        },
+        []
+    );
+
+    readonly everyStudyIdToStudy = remoteData({
+        await: () => [this.allStudies],
+        invoke: () =>
+            Promise.resolve(_.keyBy(this.allStudies.result!, s => s.studyId)),
+    });
+
+    // contains queried physical studies
+    readonly filteredPhysicalStudies = remoteData({
+        await: () => [this._session],
+        invoke: async () => {
+            const originStudies = this._session.result!.origin;
+            const everyStudyIdToStudy = this.everyStudyIdToStudy.result!;
+            return _.reduce(
+                originStudies,
+                (acc: CancerStudy[], next) => {
+                    if (everyStudyIdToStudy[next]) {
+                        acc.push(everyStudyIdToStudy[next]);
+                    }
+                    return acc;
+                },
+                []
+            );
+        },
+        default: [],
+    });
+
+    // contains queried vaild virtual studies
+    readonly filteredVirtualStudies = remoteData({
```

Comment:
queriedVirtualStudies?   perhaps just virtualStudies would be enough because we never obtain a list of all virtual studies.  also, improve comment.